packages/fx-core/src/plugins/resource/frontend/utils/environment-utils.ts

```diff
@@ -20,4 +22,43 @@ export class EnvironmentUtils {
     }
     await fs.writeFile(envFile, envs);
   }
+
+  static async updateEnvironment(
+    envFilePath: string,
+    runtimeEnv: RuntimeEnvironment,
+    aadEnv: AADEnvironment,
+    functionEnv?: FunctionEnvironment
+  ): Promise<void> {
+    let envs = await EnvironmentUtils.readEnvironments(envFilePath);
```

Comment:
Nit: The code looks good to me. Here is my preference style. The code looks good to me. But please ask @hund030 to confirm the entire logic because he is the owner for this plugin.
```
  const envs = (await EnvironmentUtils.readEnvironments(envFilePath)) ?? {};
  const addEnv = (key: string, v: string) => {
    if (v !== undefined && envs[key] === undefined) {
      envs[key] = v;
    }
  };

  addEnv(EnvironmentVariables.FuncName, functionEnv?.defaultName);
  addEnv(EnvironmentVariables.FuncEndpoint, functionEnv?.endpoint);
  addEnv(EnvironmentVariables.RuntimeEndpoint, runtimeEnv.endpoint);
  addEnv(EnvironmentVariables.StartLoginPage, runtimeEnv.startLoginPageUrl);
  addEnv(EnvironmentVariables.ClientID, aadEnv.clientId);

  await EnvironmentUtils.writeEnvironments(envFilePath, envs);
```

Reply:
Thanks for your tip